pageserver/src/tenant/storage_layer/image_layer.rs

```diff
@@ -873,15 +884,16 @@ impl ImageLayerWriterInner {
         crate::metrics::COMPRESSION_IMAGE_INPUT_BYTES_CHOSEN.inc_by(self.uncompressed_bytes_chosen);
         crate::metrics::COMPRESSION_IMAGE_OUTPUT_BYTES.inc_by(compressed_size);
 
-        let mut file = self.blob_writer.into_inner();
+        let file = self.blob_writer.into_inner(ctx).await?;
 
         // Write out the index
-        file.seek(SeekFrom::Start(index_start_blk as u64 * PAGE_SZ as u64))
-            .await?;
+        // TODO(yuchen): should we just replace BlockBuf::blocks with one big buffer?
```

Comment:
Don't see a reason why not. Please create a follow-up good_first_issue like `investigate & do if possible: replace BlockBuf::blocks with one big buffer`

Reply:
daf214e2e9e8c04df11c75d2fd1c1f603f44f9ec: created issue at https://github.com/neondatabase/neon/issues/10092